packages/disposable/src/disposeAll.js

```diff
@@ -1,9 +1,17 @@
-/** @license MIT License (c) copyright 2010-2017 original author or authors */
-import { reduce, curry2 } from '@most/prelude'
+/** @license MIT License (c) copyright 2010 original author or authors */
+import { append, concat, reduce, curry2 } from '@most/prelude'
+import { disposeNone, isDisposeNone } from './disposeNone'
 
 // Aggregate a list of disposables into a DisposeAll
-export const disposeAll = ds =>
-  new DisposeAll(ds)
+export const disposeAll = ds => {
+  const merged = ds.reduce(merge, [])
+  return merged.length === 0 ? disposeNone() : new DisposeAll(merged)
+}
+
+const merge = (ds, d) =>
```

Comment:
`merge` merges a new disposable `d` into an array of disposables `ds`.  Since, disposeNone is a noop, it can be discarded instead of adding to the array of disposables.  If `d` is a DisposeAll (which contains an array of disposables), the two arrays, `ds` and `d.disposables` can just be concatenated.